### PR TITLE
Fix Bug 7823 - Frame corner radius

### LINKED ISSUE #7823: [Bug] Frame corner radius.
### Description
After updating from v3.6.0.344457 to v4.2.0.848062 frame cease to cut corners of child elements.
### Steps to Reproduce

1. Create frame with any corner radius and NOT ZERO padding.
2. Create one more frame as a child of the first with any corner radius and ZERO `padding.`
3. Into the second frame insert boxview(can be any view)

### Expected Behavior
Child elements has cut corners.
### Actual Behavior
Child elements don't has cut corners

### Code for screenshot

    <StackLayout>
        <Frame CornerRadius="5" BackgroundColor="Red" Padding="10">
            <Frame CornerRadius="10" BackgroundColor="Blue" Padding="0">
                <BoxView BackgroundColor="Green" HeightRequest="100" WidthRequest="100"/>
            </Frame>
        </Frame>
    </StackLayout>

### Screenshots
photo_2019-10-04_16-53-25

### PR DESCRIPTION
**Pull request by felipebaltazar**
### Description of Change ###

> Important! This is a #7966 retarget to 4.2.0 branch 

I created a validation at "UpdateClippedToBounds" event, to check if "IsClippedToBounds" is set by the user to use that value or check if the "CornerRadius" is greater than 0.

### Issues Resolved ### 
- fixes #7823 

### API Changes ###
 None

### Platforms Affected ### 
- Core/XAML (all platforms)

### Behavioral/Visual Changes ###

The frame will cut corners of child elements when a corner radius greater than 0 and padding equals 0. (If user doesn't set "IsClippedToBounds = false")

### Before/After Screenshots ### 

- Before

<p align="center">
	<kbd>
		before screenshot
	</kbd>
</p>

- After
<p align="center">
	<kbd>
		image
	</kbd>
</p>

### Testing Procedure ###

1. Create a frame with any corner radius and NOT ZERO padding.
1. Create one more frame as a child of the first with any corner radius and ZERO padding.
1. Into the second frame insert boxview(can be any view)

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jfversluis** [comment]
UI tests failing

> System.NullReferenceException : Object reference not set to an instance of an object  at Xamarin.Forms.Controls.Issues.Issue7823.AssertIsClipped (System.Boolean expected) [0x00048] in <536d6b9c4c724c62aa9b2f628651fe74>:0   at Xamarin.Forms.Controls.Issues.Issue7823.Issue7823TestIsClippedIssue () [0x00032] in <536d6b9c4c724c62aa9b2f628651fe74>:0   at (wrapper managed-to-native) System.Reflection.MonoMethod:InternalInvoke (System.Reflection.MonoMethod,object,object[],System.Exception&)  at System.Reflection.MonoMethod.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x00032] in <48b95f3df5804531818f80e28ec60191>

File: Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7823.cs [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+using NUnit.Framework;
+using Xamarin.UITest;
+using Xamarin.Forms.Core.UITests;
+using Xamarin.UITest.Queries;
+#endif
+
+namespace Xamarin.Forms.Controls.Issues
+{
+	[Preserve(AllMembers = true)]
+	[Issue(IssueTracker.Github, 7823, "[Bug] Frame corner radius.", PlatformAffected.All)]
```
Comment (jfversluis) [inline Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7823.cs]:
```suggestion
	[Issue(IssueTracker.Github, 7823, "[Bug] Frame corner radius.", PlatformAffected.Android)]
```

Comment (felipebaltazar) [reply Xamarin.Forms.Controls.Issues/Xamarin.Forms.Controls.Issues.Shared/Issue7823.cs]: i forgot to change this, sorry...

**Comment by felipebaltazar** [comment]
humm... I'm very confused about why is the target branch...
I've created this PR targeting 4.2.0, that's was solicited at [previous PR](https://github.com/xamarin/Xamarin.Forms/pull/7966)

Before the base branch changed, I solved some conflicts but it was retargeted again... 🤔
So, I will work at master, 4.3 or 4.2?

image

**Comment by rmarinho** [comment]
@samhouts can you help me figure this one? should be 4.2.0 right?

**Comment by samhouts** [comment]
Since 4.3.0 is our new stable version, we're unlikely to support 4.2.0 except for critical items. If this were a regression on 4.3.0, that would be the recommended target branch. Since this is an earlier regression (possibly somewhere between 4.0.0 and 4.2.0), we should probably just consider it a "bug" to be safe and put it in the latest prerelease. At this point, that is the 4.4.0 branch. 

Phew! Sounds more confusing that it should be. :)

**Comment by felipebaltazar** [comment]
> Since 4.3.0 is our new stable version, we're unlikely to support 4.2.0 except for critical items. If this were a regression on 4.3.0, that would be the recommended target branch. Since this is an earlier regression (possibly somewhere between 4.0.0 and 4.2.0), we should probably just consider it a "bug" to be safe and put it in the latest prerelease. At this point, that is the 4.4.0 branch.
> 
> Phew! Sounds more confusing that it should be. :)

It's ok! @samhouts thank you for the explanation!

**Comment by felipebaltazar** [comment]
I've rebased this PR to 4.4.0 and resolved some "reversions" that I did because the target was changed...

**Comment by rmarinho** [comment]
Failing tests not related

**Comment by felipebaltazar** [comment]
> The API27 tests are actually legit
> 
> > Expected: True  But was:  null  at Xamarin.Forms.Controls.Issues.Issue7823.AssertIsClipped (System.Boolean expected) [0x00080] in :0   at Xamarin.Forms.Controls.Issues.Issue7823.Issue7823TestIsClippedIssue () [0x00032] in :0

I've updated the android test's and checked AP27.. seems like ok

**Comment by felipebaltazar** [comment]
I've been fixed API 19 test and fast renderer logic

File: Xamarin.Forms.Platform.Android/FastRenderers/FrameRenderer.cs [hunk 2]
```diff
@@ -236,13 +236,16 @@ protected virtual void OnElementPropertyChanged(object sender, PropertyChangedEv
 				UpdateClippedToBounds();
 		}
 
-		void UpdateClippedToBounds() => this.SetClipToOutline(Element.IsClippedToBounds);
+		void UpdateClippedToBounds()
```
Comment (jfversluis) [inline Xamarin.Forms.Platform.Android/FastRenderers/FrameRenderer.cs]: It's not implemented for the regular renderer?

Comment (felipebaltazar) [reply Xamarin.Forms.Platform.Android/FastRenderers/FrameRenderer.cs]: Hrs, regular renderer uses VisualTracker do apply this property.

**Pull request merged**